stress/n_workers_reputers__stress_test.go

```diff
@@ -272,16 +273,183 @@ func InsertReputerBulk(m TestMetadata,
+	workersMax := lookupEnvInt(m, "WORKERS_MAX", 10000)
+	topicsPerEpoch := lookupEnvInt(m, "TOPICS_PER_EPOCH", 0)
+	topicsMax := lookupEnvInt(m, "TOPICS_MAX", 100)
+
+	fmt.Println("Reputers per epoch: ", reputersPerEpoch)
+	fmt.Println("Reputers max: ", reputersMax)
+	fmt.Println("Workers per epoch: ", workersPerEpoch)
+	fmt.Println("Workers max: ", workersMax)
+	fmt.Println("Topics per epoch: ", topicsPerEpoch)
+	fmt.Println("Topics max: ", topicsMax)
+
 	approximateBlockTimeSeconds := getApproximateBlockTimeSeconds(m)
 	fmt.Println("Approximate block time seconds: ", approximateBlockTimeSeconds)
 
-	const stakeToAdd uint64 = 90000
-	const topicFunds int64 = 1000000
+	iterationTimeSeconds := time.Duration(epochLength) * approximateBlockTimeSeconds * iterationsInABatch
+	topicCount := 0
+	topicFunderCount := 0
+
+	topicFunderAddresses := make([]string, 0)
+
+	for topicFunderIndex := 0; topicFunderIndex < topicsMax; topicFunderIndex++ {
+		topicFunderAccountName := getTopicFunderAccountName(topicFunderIndex)
+		topicFunderAccount, _, err := m.n.Client.AccountRegistry.Create(topicFunderAccountName)
+		if err != nil {
+			fmt.Println("Error creating funder address: ", topicFunderAccountName, " - ", err)
+			continue
+		}
+		topicFunderAddress, err := topicFunderAccount.Address(params.HumanCoinUnit)
+		if err != nil {
+			fmt.Println("Error creating funder address: ", topicFunderAccountName, " - ", err)
+			continue
+		}
+		topicFunderAddresses = append(topicFunderAddresses, topicFunderAddress)
+	}
+
+	err := fundAccounts(m, m.n.FaucetAcc, m.n.FaucetAddr, topicFunderAddresses, 1e9)
+	if err != nil {
+		fmt.Println("Error funding funder accounts: ", err)
+	} else {
+		fmt.Println("Funded", len(topicFunderAddresses), "funder accounts.")
+	}
+
+	getTopicFunder := func() (string, cosmosaccount.Account) {
+		topicFunderAccountName := getTopicFunderAccountName(topicFunderCount)
+		topicFunderCount++
+		topicFunderAccount, err := m.n.Client.AccountRegistry.GetByName(topicFunderAccountName)
+		if err != nil {
+			fmt.Println("Error getting funder account: ", topicFunderAccountName, " - ", err)
+			return "", topicFunderAccount
+		}
+		topicFunderAddress, err := topicFunderAccount.Address(params.HumanCoinUnit)
+		if err != nil {
+			fmt.Println("Error getting funder address: ", topicFunderAccountName, " - ", err)
+			return topicFunderAddress, topicFunderAccount
+		}
+		return topicFunderAddress, topicFunderAccount
+	}
+
+	workerCount := 1
+	reputerCount := 1
+
+	if topicsPerEpoch == 0 {
+		topicFunderAddress, topicFunderAccount := getTopicFunder()
+		WorkerReputerLoop(m, topicFunderAddress, topicFunderAccount, workerCount, reputerCount)
+		topicCount++
+	} else {
+		for {
+			startIteration := time.Now()
+
+			for j := 0; j < topicsPerEpoch && topicCount < topicsMax; j++ {
+				topicFunderAddress, topicFunderAccount := getTopicFunder()
+				go WorkerReputerLoop(m, topicFunderAddress, topicFunderAccount, workerCount, reputerCount)
```

Comment:
Not clear to me why initialWorkerCount, initialReputerCount (used here as workerCount, reputerCount) increase in every outside iteration (4 lines above)
Is it so the new topics "catch up" with the number of workers and reputers that the previous topics have?

Reply:
Yes, it's so the new topics catch up.